frontend/src/core/api/types.js

```diff
@@ -32,6 +43,7 @@ export type Entity = {|
     +project: Object,
     +source: Array<Array<string>> | Object,
     +translation: Array<EntityTranslation>,
+    +translation_comments: Array<TranslationComments>,
```

Comment:
I'd move this property under `HistoryTranslation` and rename it to `comments`. The comments data should be loaded and stored together with the rest of the history panel data.